perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -3057,56 +3057,51 @@ private void updateExistingMembersWhileSynchronization(PerunSession sess, Group
 			}
 
 			//Set correct member Status
-			// If the member has expired or disabled status, try to expire/validate him (depending on expiration date)
-			if (richMember.getStatus().equals(Status.DISABLED) || richMember.getStatus().equals(Status.EXPIRED)) {
+			List<Status> affectedStatuses = new ArrayList<>();
```

Comment:
We can make this list ``final static`` variable of the class. Like this its constructed for each member in the cycle.

Reply:
Moved as static list for the whole class.